File: PadraoCriacionais/Singleton/Singleton.py
class Singleton:
    instance = None

    def __new__(cls):
        if cls.instance is None:
            cls.instance = super().__new__(cls)
            cls.instance.value = 0
        return cls.instance
    
    def increment(self):
        self.value += 1
        print(f"Value: {self.value}")



# Cliente
s1 = Singleton()
s2 = Singleton()
s3 = Singleton()
s4 = Singleton()

print (s1 == s2)    # True

s1.increment()
s2.increment()
s3.increment()
s4.increment()

File: PadraoCriacionais/Singleton/Singleton.js
class Singleton{
    constructor(){
        if(Singleton.instance == null){
            Singleton.instance = this;
            this.value = 0;
        }
        return Singleton.instance;
    }

    increment(){
        this.value += 1;
        console.log(`Value: ${this.value}`);
    }
}

// Cliente
const s1 = new Singleton();
const s2 = new Singleton();
const s3 = new Singleton();
const s4 = new Singleton();

console.log(s1 == s2);      // True

s1.increment();
s2.increment();
s3.increment();
s4.increment();